client.go

```diff
@@ -380,6 +380,10 @@ func (s *Client) resumeSwaps(ctx context.Context,
 			continue
 		}
 
+		// Store the swap's abandon channel so that the client can
+		// abandon the swap by providing the swap hash.
+		s.abandonChans[swap.hash] = swap.abandonChan
```

Comment:
Locking is also needed here as this function runs in its own goroutine and the executor might already be active.